server/api_conversation.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/pluginapi"
+
+	"github.com/gin-gonic/gin"
+)
+
+func (p *Plugin) handlePostConversation(c *gin.Context) {
+	userID := c.GetHeader("Mattermost-User-Id")
+
+	// We only allow bots to use this API handler for the time being.
+	if _, err := p.pluginAPI.Bot.Get(userID, false); errors.Is(err, pluginapi.ErrNotFound) {
+		c.AbortWithError(http.StatusForbidden, errors.New("forbidden"))
+		return
+	} else if err != nil {
+		c.AbortWithError(http.StatusInternalServerError, fmt.Errorf("failed to get bot: %w", err))
+		return
+	}
+
+	var conv []*model.Post
```

Comment:
I thought a simple slice would be clean enough, but I'm happy to consider a more advanced approach, such as passing some sort of custom object that wraps the data we need.

Reply:
Maybe we want to just wrap that in a struct so we can introduce other parameters if need be without a breaking change.

Reply:
Yep, that's probably a healthier choice in the long run. Will update :+1: